src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutPresenter.cs

```diff
@@ -1,13 +1,958 @@
+
+			// We should wrap around at the bottom or the top of the presenter if the user isn't using a gamepad or remote.
+			var shouldWrap = parentFlyout != null ? (parentFlyout.InputDeviceTypeUsedToOpen != FocusInputDeviceKind.GameController) : true;
+
+			var nCount = Items.Size;
+
+			// Determine direction of index movement based on the Up/Down key.
+			var deltaIndex = shouldCycleDown ? 1 : -1;
+
+			// Set index by moving deltaIndex amount from the current focused item.
+			var index = m_iFocusedIndex + deltaIndex;
+
+			// We have two locations where we want to wrap, so we'll encapsulate the wrapping behavior in a function object that we can call.
+			int wrapIndexIfNeeded(int indexToWrap)
+			{
+				if (shouldWrap)
+				{
+					if (indexToWrap < 0)
+					{
+						indexToWrap = (int)(nCount) - 1;
+					}
+					else if (indexToWrap >= (int)(nCount))
+					{
+						indexToWrap = 0;
+					}
+				}
+
+				return indexToWrap;
+			}
+
+			// If there is no item focused right now, then set index to 0 for Down key or to n-1 for Up key.
+			// Otherwise, if we should be wrapping, we'll do an initial check for whether we should wrap before we enter the loop.
+			if (m_iFocusedIndex == -1)
+			{
+				index = shouldCycleDown ? 0 : (int)(nCount) - 1;
+
+				// If the focused index is -1, then our value of -1 for originalFocusedIndex will not successfully stop the loop.
+				// In this case, we'll make originalFocusedIndex one step in the opposite direction from the initial index,
+				// so that way the loop will go all the way through the list of items before stopping.
+				originalFocusedIndex = wrapIndexIfNeeded(index - deltaIndex);
+			}
+			else
+			{
+				index = wrapIndexIfNeeded(index);
+			}
+
+			// We need to examine all items with indices [0, m_iFocusedIndex) or (m_iFocusedIndex, nCount-1] for Down/Up keys.
+			// While index is within the range, we keep going through the item list until we are successfully able to focus an item,
+			// at which point we update the m_iFocusedIndex and break out of the loop.
+			while (0 <= index && index < (int)(nCount))
+			{
+				var spItemAsDependencyObject = Items[index] as DependencyObject;
+
+				var isFocusable = false;
+				// We determine whether the item is a focusable MenuFlyoutItem or MenuFlyoutSubItem because we want to exclude MenuSeparators here.
+				var spItem = spItemAsDependencyObject as MenuFlyoutItem;
+				if (spItem != null)
+				{
+					isFocusable = spItem.IsFocusable;
+				}
+				else
+				{
+					MenuFlyoutSubItem spSubItem;
+					spSubItem = spItemAsDependencyObject as MenuFlyoutSubItem;
+					if (spSubItem != null)
+					{
+						isFocusable = spSubItem.IsFocusable;
+					}
+				}
+
+				// If the item is focusable, move the focus to it, update the m_iFocusedIndex and break out of the loop.
+				if (isFocusable)
+				{
+					var spSubItem = spItemAsDependencyObject as Control;
+					if (spSubItem != null)
+					{
+						var wasFocused = spSubItem.Focus(focusState);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'wasFocused'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)